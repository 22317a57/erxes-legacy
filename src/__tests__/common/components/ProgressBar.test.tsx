import { mount, shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import ProgressBar from '../../../modules/common/components/ProgressBar';

describe('ProgressBar component', () => {
  const defaultProps = {
    percentage: 50
  };

  test('renders shallow successfully', () => {
    shallow(<ProgressBar {...defaultProps} />);
  });

  test('renders mount successfully with default value', () => {
    const wrapper = mount(<ProgressBar {...defaultProps} />);
    const props = wrapper.props();

    expect(props).toMatchObject(defaultProps);
  });
  test('snapshot matches', () => {
    const rendered = renderer
      .create(<ProgressBar {...defaultProps} />)
      .toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
