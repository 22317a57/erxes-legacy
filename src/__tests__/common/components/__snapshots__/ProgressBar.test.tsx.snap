// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component snapshot matches 1`] = `
<div
  className="sc-bwzfXH fbeMma"
>
  <div
    className="sc-htpNat fWIABv"
    style={
      Object {
        "width": "50%",
      }
    }
  />
  <div
    className="sc-bdVaJa kOEvUl"
  />
</div>
`;
